tests/run.py

```diff
@@ -2,41 +2,134 @@
+    for fight_style in FIGHT_STYLES:
+        grp = TestGroup('{}/{}/covenants'.format(profile, fight_style),
+                        fight_style=fight_style, profile=path)
+        tests.append(grp)
+
+        # Test covenants
+        for covenant in Covenant.COVENANTS:
+            Test(covenant.full_name, group=grp,
+                 args=[('covenant', covenant.simc_name), ('level', 60)])
+            # Add conduits specific to selected covenant
+            for conduit in conduits:
+                if len(conduit.covenants) == 1 and covenant == conduit.covenants[0]:
+                    rank = random.choice(conduit.ranks) + 1
+                    soulbind_argument = '{}:{}'.format(conduit.id, rank)
+                    Test('{} - {} ({}) Rank {}'.format(covenant.full_name, conduit.full_name, conduit.id, rank), group=grp, args=[
+                        ('covenant', covenant.simc_name), ('level', 60), ('soulbind', soulbind_argument)])
+
+        # test conduits available for all 4 covenants independent from covenant selection.
+        for conduit in conduits:
+            if len(conduit.covenants) == 4:
+                rank = random.choice(conduit.ranks) + 1
+                soulbind_argument = '{}:{}'.format(conduit.id, rank)
+                Test('{} ({}) Rank {}'.format(conduit.full_name, conduit.id, rank), group=grp, args=[
+                    ('level', 60), ('soulbind', soulbind_argument)])
+
+
+def test_trinkets(klass: str, path: str):
+    spec = WowSpec.get_wow_spec_from_combined_simc_name(klass)
+    trinkets = Trinket.get_trinkets_for_spec(spec)
     fight_style = args.trinkets_fight_style
     grp = TestGroup('{}/{}/trinkets'.format(profile, fight_style),
                     fight_style=fight_style, profile=path)
     tests.append(grp)
     for trinket in trinkets:
         Test('{} ({})'.format(trinket.name, trinket.item_id), group=grp, args=[
-                ('trinket1', '{},id={},ilevel={}'.format(trinket.name, trinket.item_id, trinket.min_itemlevel))])
+            ('trinket1', '"{}",id={},ilevel={}'.format(trinket.name, trinket.item_id, trinket.min_itemlevel))])
+
+
+def test_legendaries(klass: str, path: str):
+    spec = WowSpec.get_wow_spec_from_combined_simc_name(klass)
+    legendaries = Legendary.get_legendaries_for_spec(spec)
+    for fight_style in FIGHT_STYLES:
+        grp = TestGroup('{}/{}/legendaries'.format(profile, fight_style),
+                        fight_style=fight_style, profile=path)
+        tests.append(grp)
+        for legendary in legendaries:
+            Test('{} ({} / {})'.format(legendary.full_name, legendary.id, legendary.bonus_id), group=grp, args=[
+                ('trinket1', '"{}",bonus_id={}'.format(legendary.full_name, legendary.bonus_id))])
+
+
+def test_soulbinds(klass: str, path: str):
+    fight_style = args.soulbind_fight_style
+    for covenant in Covenant.COVENANTS:
+        grp = TestGroup('{}/{}/soulbinds'.format(profile, fight_style),
```

Comment:
Maybe move the covenant to the group name (considering TestGroups are per covenant)?
Everyone will know their soulbind names (if not soulbind abilities) in like a week anyway.
Should also help with test name lengths a bit.